TicketsDemo.Domain/DefaultImplementations/TicketService.cs

```diff
@@ -44,8 +47,16 @@ public Ticket CreateTicket(int reservationId, string fName, string lName)
                 Status = TicketStatusEnum.Active,
                 PriceComponents = new List<PriceComponent>()
             };
+            var teaCoffeeBedParametrs = new TeaCoffeeBedParametrs()
+            {
+                ticket = newTicket,
```

Comment:
as mentioned before, ticket being a part of TicketCreationParameters introduce a mess into the code.
Here you creating the ticket object 'before' calculating the price.

If you get rid of 'ticket' property - you'll not need to create a ticket 'before' calculating the price